### ISSUE TITLE
VerseRef constructor accepts invalid input

### ISSUE THREAD

**Issue by Nateowami** [sillsdev/scripture#18]
This throws an exception, as expected:
``` ts
new VerseRef('1PE 3')
```
This does not throw an exception:
```ts
new VerseRef('1PE 3:6=&*(&^&*$%&^%^asdfoajsdfjalsdhfnjk56asdhf')
```

**Comment by irahopkinson** [comment]
Indeed, the first one throws an _Invalid reference_ exception because the reference is incomplete without also specifying the verse.

For the second one it is less clear. You could argue that it is following the good practice of being tolerant of input since `verseNum` does get parsed to the number `6`, however `verse` is `"6=&*(&^&*$%&^%^asdfoajsdfjalsdhfnjk56asdhf"`. I agree that's not good overall.

Given that this is actually following the original behavior of the C#, I'm a little reticent to "fix" it here. Perhaps we should repost this issue to the C# repo? Is there an existing issue that fixing this in the TS version would help with or is this just messy behavior for now?

**Comment by Nateowami** [comment]
> You could argue that it is following the good practice of being tolerant of input since `verseNum` does get parsed to the number `6`, however `verse` is `"6=&*(&^&*$%&^%^asdfoajsdfjalsdhfnjk56asdhf"`.

Correct. However, from my perspective this behavior is useless, since it hasn't actually parsed the verse range. I would like to be able to convert a verse ref into a list of discrete verses. If that can't actually be done, I don't think the software should accept it as a reference. Maybe there's a use-case I'm not aware of, but Scripture Forge would want to be able to highlight verses associated with a question, which is associated with a verse ref.

**Comment by irahopkinson** [comment]
Perhaps I'm missing something but `6` appears to be the only verse range that I can parse from `"6=&*(&^&*$%&^%^asdfoajsdfjalsdhfnjk56asdhf"`. What is your expected outcome that `VerseRef` should do with this?

**Comment by Nateowami** [comment]
It seems to accept literally anything as a verse range, and then not parse the verse range, except to try to get the first number out of it. VerseRef.tryParse returns an object with `success: true` almost no matter how badly formed the verse range part is.

The only workaround for this I can think of is to take the verse range part, and then try to parse it by splitting on `,`, `-`, and maybe a couple of other things, but that seems like something this library ought to do itself.
